### PR TITLE
feat: hid expired quests and expired collections

### LINKED ISSUE #769: Only show active quests in the collection section
In the collection section, we need to hide quests that are no longer available. Only show active quests.

https://www.loom.com/share/be52667b4de04b12aa0d0cc3d06ddc8e

### PR DESCRIPTION
**Pull request by Ayoazeez26**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

Resolves: #769

## Summary by CodeRabbit

- **New Features**
	- Implemented a filtering mechanism for quest categories, enhancing the responsiveness and accuracy of displayed information.
	- Updated the quest completion display to accurately reflect the count of ongoing quests.

- **Bug Fixes**
	- Corrected the logic for displaying the number of completed quests to ensure alignment with the actual data.

- **Tests**
	- Added unit tests for the new filtering function to validate its functionality across various scenarios.

### CODE REVIEW

File: utils/quest.ts [hunk 1]
```diff
@@ -44,3 +44,11 @@ export const getClaimableQuests = (
     }));
   return questIdsInBoostClaim as ClaimableQuestDocument[];
 };
+
+export const filterQuestCategories = (categories: QuestCategory[]): QuestCategory[] => {
```
Comment (fricoben) [inline utils/quest.ts:48]: I feel like using the first `.map` here is not needed

```ts
export const filterQuestCategories = (categories: QuestCategory[]): QuestCategory[] => {
  return categories.filter((category) => {
    category.quests = category.quests.filter((quest) => !quest.expired);
    return category.quests.length > 0;
  });
}
```

**Review by fricoben — CHANGES_REQUESTED** [review]
I proposed a small refactoring of the filtering function

**Review by fricoben — CHANGES_REQUESTED** [review]
## Build Error 

Build Error my proposed function might be not equivalent I'll let you refactor. 

```ts
Type error: This comparison appears to be unintentional because the types '"Quiz" | "Discord" | "Custom" | "Domain" | "Balance" | "TwitterFw" | "TwitterRw"' and '"None"' have no overlap.
  283 |     const updatedTasks = filteredSteps.filter((step, index) => {
  284 |       return (
> 285 |         step?.type !== "None" &&
      |         ^
  286 |         intialSteps[index]?.type !== "None" &&
  287 |         step?.data?.id &&
  288 |         step?.data?.id !== 0 &&
Error: Command "npm run build" exited with 1
```

File: app/admin/quests/dashboard/[questId]/page.tsx [hunk 2]
```diff
@@ -282,7 +282,6 @@ export default function Page({ params }: QuestIdProps) {
 
     const updatedTasks = filteredSteps.filter((step, index) => {
       return (
```
Comment (Ayoazeez26) [inline app/admin/quests/dashboard/[questId]/page.tsx:284]: The filteredSteps already eliminates steps with the type of "None" hence the error

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**